library/Zend/Config/Yaml.php

```diff
@@ -289,7 +289,7 @@ protected static function _decodeYaml($currentIndent, &$lines)
     {
         $config   = array();
         $inIndent = false;
-        while (list($n, $line) = each($lines)) {
+        foreach ($lines as $n => $line) {
```

Comment:
This actually breaks the component, somehow the result is not the same.

Reply:
Would you be so kind to provide an example of this problem? Thank you.

Reply:
Yeah, this breaks ZF1 tests, we experienced the same. It's because `each` moves the array pointer on `$lines` which is passed by reference, but `foreach` does not (there's also some pointer movement below with `prev`).

Correct patch here (we believe, tests passed): https://github.com/diablomedia/zf1/commit/ba8cf7aaf6542b7aadee79146709302d65e85bdd#diff-9235a8f0d83484e2adcf78f7ebffe2caL292